src/Simplex_tree/include/gudhi/Simplex_tree.h

```diff
@@ -1347,52 +1366,317 @@ class Simplex_tree {
+    * an existing simplex.
+    */
+  void insert_edge_as_flag(  Vertex_handle                   u
+                           , Vertex_handle                   v
+                           , Filtration_value                fil
+                           , int                             dim_max
+                           , std::vector<Simplex_handle>&    added_simplices)
+  {
+    /**
+     * In term of edges in the graph, inserting edge `[u,v]` only affects
+     * the subtree rooted at @p u.
+     *
+     * For a new node with label @p v, we first do a local expansion for
+     * computing the children of this new node, and then a standard expansion
+     * for its children.
+     * Nodes with label @p v (and their subtrees) already in the tree
+     * do not get affected.
+     *
+     * Nodes with label @p u get affected only if a Node with label @p v is in their same
+     * siblings set.
+     * We then try to insert "ponctually" @p v all over the subtree rooted
+     * at `Node(u)`. Each insertion of a Node with @p v label induces a local
+     * expansion at this Node (as explained above) and a sequence of "ponctual"
+     * insertion of `Node(v)` in the subtree rooted at sibling nodes of the new node,
+     * on its left.
+     */
+
+    static_assert(Options::link_nodes_by_label, "Options::link_nodes_by_label must be true");
+
+    if (u == v) { // Are we inserting a vertex?
+      auto res_ins = root_.members().emplace(u, Node(&root_,fil));
+      if (res_ins.second) { //if the vertex is not in the complex, insert it
+        added_simplices.push_back(res_ins.first); //no more insert in root_.members()
+        update_simplex_tree_after_node_insertion(res_ins.first);
+        if (dimension_ == -1) dimension_ = 0;
+      }
+      return; //because the vertex is isolated, no more insertions.
+    }
+    // else, we are inserting an edge: ensure that u < v
+    if (v < u) { std::swap(u,v); }
+
+    //Note that we copy Simplex_handle (aka map iterators) in added_simplices
+    //while we are still modifying the Simplex_tree. Insertions in siblings may
+    //invalidate Simplex_handles; we take care of this fact by first doing all
+    //insertion in a Sibling, then inserting all handles in added_simplices.
+
+    //check whether vertices u and v are in the tree. If not, return an error.
+    auto sh_u = root_.members().find(u);
+    GUDHI_CHECK(sh_u != root_.members().end() &&
+          root_.members().find(v) != root_.members().end(),
+          std::invalid_argument(
+                  "Simplex_tree::insert_edge_as_flag - inserts an edge whose vertices are not in the complex")
+                );
+    GUDHI_CHECK(!has_children(sh_u) ||
+          sh_u->second.children()->members().find(v) == sh_u->second.children()->members().end(),
+          std::invalid_argument(
+                  "Simplex_tree::insert_edge_as_flag - inserts an already existing edge")
+                );
```

Comment:
Surrounding this code by `#ifdef GUDHI_DEBUG` would remove the `warning: sh_u defined but not used`. 
```suggestion
#ifdef GUDHI_DEBUG
    //check whether vertices u and v are in the tree. If not, return an error.
    auto sh_u = root_.members().find(u);
    GUDHI_CHECK(sh_u != root_.members().end() &&
          root_.members().find(v) != root_.members().end(),
          std::invalid_argument(
                  "Simplex_tree::insert_edge_as_flag - inserts an edge whose vertices are not in the complex")
                );
    GUDHI_CHECK(!has_children(sh_u) ||
          sh_u->second.children()->members().find(v) == sh_u->second.children()->members().end(),
          std::invalid_argument(
                  "Simplex_tree::insert_edge_as_flag - inserts an already existing edge")
                );
#endif
```

`[[maybe_unused]] auto sh_u ...` would also work for me.